include/nbl/builtin/hlsl/shapes/beziers.hlsl

```diff
@@ -40,89 +40,204 @@ namespace shapes
         numRealValues = 2;
         return float2(sqrtD - b, sqrtD + b);
     }
-
+    
+    template<typename float_t>
     struct QuadraticBezier
     {
-        float2 A;
-        float2 B;
-        float2 C;
+        using float2_t = vector<float_t, 2>;
+        using float3_t = vector<float_t, 3>;
 
-        static QuadraticBezier construct(float2 a, float2 b, float2 c)
+        struct ArcLengthPrecomputedValues
```

Comment:
I think you should remove this and all the arc len stuff in quadratic bezier struct?

Reply:
done